eng/pipelines/pr.job.yml

```diff
@@ -0,0 +1,104 @@
+parameters:
+- name: displayName
+- name: osName
+- name: agentPool
+  type: string
+  default: Azure Pipelines
+- name: agentDemands
+  type: string
+  default: none
+- name: vmImage
+  type: string
+  default: none
+- name: testType
+- name: testTargetFramework
+- name: timeoutInMinutes
+  type: number
+  default: 15
+
+jobs:
+- job:
+  displayName: ${{ parameters.displayName }}
+  timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+  variables:
+    - name: TestType
+      value: ${{ parameters.testType }}
+    - name: TestTargetFramework
+      value: ${{ parameters.testTargetFramework }}
+    - name: TestResultsDirectory
+      ${{ if eq(parameters.osName, 'Windows') }}:
+        value: $(Agent.TempDirectory)\TestResults
+      ${{ else }}:
+        value: $(Agent.TempDirectory)/TestResults
```

Comment:
Windows supports `/`. Is there a reason we can't use that everywhere, and avoid a lot of these "if windows .. else .." blocks?

Reply:
The value here gets passed to `dotnet test` and that appears to not support a single value on Windows.  I might be able to get rid of this logic as I was originally running multiple `dotnet test` instances for each test assembly and then publishing the results as merged which requires an extra pipeline task.  The current iteration just runs all tests under a single `dotnet test` invocation so maybe I can get away without needing this.  I'll test it tomorrow...

Reply:
And it turns out it wasn't necessary due to other changes, thanks for pointing it out.  I've removed it and the publish test result task to make the job yaml even simpler!